Sources/TuistAnalytics/Utilities/TuistAnalyticsDispatcher.swift

```diff
@@ -18,16 +18,16 @@ public struct TuistAnalyticsDispatcher: AsyncQueueDispatching {
 
     public var identifier = TuistAnalyticsDispatcher.dispatcherId
 
-    public func dispatch(event: AsyncQueueEvent, completion: @escaping () throws -> Void) throws {
+    public func dispatch(event: AsyncQueueEvent, completion: @escaping () async throws -> Void) throws {
         guard let commandEvent = event as? CommandEvent else { return }
 
         Task.detached {
```

Comment:
Now that we're marking the containing method as `async` I think we'll want to just use `Task` in these places instead of `Task.detached`.   Using `detached` breaks the relationship between this task and the calling method so that if one were to `cancel()` the call to `dispatch(event:...)` this task would continue to run.  There were some other usages of `Task.detached` and those likely should also be changed once their containing methods are also `async`

Reply:
Addressed!